Edit README

tajmone opened the pull request:
- Fix a couple of typos.
- Convert inline links to reference-style links.
- Change inline-code delimiters to single backticks.
- Join split paragraphs for better diffing on changes and document
  uniformity.
- Visually format tables.

GrenDrake commented on README.md:
```diff
@@ -39,44 +33,39 @@ glulx-assemble -dump_tokens basic.ga basic.ulx
 ### Instructions
 
-An instruction statement consists of an opcode mnemonic followed by zero or more operands. For an index of opcodes and their expected operands, check the [relevant part of the glulx spec](https://www.eblong.com/zarf/glulx/glulx-spec_2.html). There are several types of values that an operand can have, summed up in the table below:
+An instruction statement consists of an opcode mnemonic followed by zero or more operands. For an index of opcodes and their expected operands, check the [relevant part of the glulx spec]glulx-spec_2.html). There are several types of values that an operand can have, summed up in the table below:
```
I think you left a bit of the original URL hanging on here.

GrenDrake commented on README.md:
````diff
@@ -111,9 +100,10 @@ A few constants are automatically defined. These are *_RAMSTART*, *_EXTSTART*, a
 ```
 
-**.function**: Adds a function header to the output file. The directive may be immediately followed by ```stk``` to specify that the arguments to this function should be passed on the stack rather than copied into the local variables (see the [glulx spec](https://www.eblong.com/zarf/glulx/glulx-spec_1.html#s.6.2) for details). This is followed by the local variable specification. Locals may be sp ...
+**.function**: Adds a function header to the output file. The directive may be immediately followed by `stk` to specify that the arguments to this function should be passed on the stack rather than copied into the local variables (see the [glulx spec][glulx spec 6.2] for details). This is followed by the local variable specification. Locals may be specified by either a positive integer (which wil ...
````
It may just be my unfamiliarity with a lot of Markdown syntax, but is there a reason you've got two [glulx spec] in there? Only one actually seems to show up in the final result.

tajmone commented:
> I think you left a bit of the original URL hanging on here.

Opps! Sorry about that, it must have been an accidental undo operation.

I've fixed it by amending the previous commit and force pushing it, so we keep a clean repo history without bad commits.

> It may just be my unfamiliarity with a lot of Markdown syntax, but is there a reason you've got two `[glulx spec]` in there? Only one actually seems to show up in the final result.

No, that's a [reference style link] where the first brackets are the link text, while the second pair of brackets is the actual reference link (`[text][reference link label]`):

```markdown
[glulx spec][glulx spec 6.2]

[glulx spec 6.2]: https://www.eblong.com/zarf/glulx/glulx-spec_1.html#s.6.2 "Read Glulx official specs on this topic"
```

I've added the ` 6.2` for the section number so that the reference link label `glulx spec` can be used to point to different sections of the spec in the future, and the reference `[glulx spec]` can be preserved to point to Glulx Spec TOC.

[reference style link]: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

<!-- EOF -->

GrenDrake commented:
Looks like there's still a stray parenthesis on that first item.

I wasn't aware of that feature for the second item. I don't actually use Markdown that much and tend not to be familiar with a lot of the features available.

tajmone commented:
> Looks like there's still a stray parenthesis on that first item.

Amended!

I've also added `*.exe` to the `.gitignore` patterns so that compiled binaries are not seen as unstaged files.
